types/types.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"log"
+	"sync"
+)
+
+// Runner is an interface which is implemented by all proxies
+type Runner interface {
+	Run() error
+	Cancel() error
+	Reload() error
+}
+
+// Server governs a slice of Runners
+type Server struct {
+	Servers []Runner
+	wg      sync.WaitGroup
+	Done    chan error
+}
+
+func (s *Server) Run() error {
+	var err error
+	for i := range s.Servers {
+		s.wg.Add(1)
+		go func(i int) {
+			defer s.wg.Done()
+			err = s.Servers[i].Run()
+			if err != nil {
+				s.Done <- err
+				log.Println(err)
+				err = s.Servers[i].Cancel()
+				if err != nil {
+					log.Println(err)
+				}
+			}
+		}(i)
+	}
+	s.wg.Wait()
+	return err
+}
+
+func (s *Server) Cancel() error {
+	var err error
+	defer close(s.Done)
+	for i := range s.Servers {
+		err = s.Servers[i].Cancel()
+		if err != nil {
+			s.Done <- err
+			log.Println(err)
+		}
+	}
+	return err
+}
+
+func (s *Server) Reload() error {
+	return nil
+}
+
+func NewServer() *Server {
+	s := make([]Runner, 0)
+	d := make(chan error)
+	return &Server{
+		Servers: s,
+		Done:    d,
+	}
+}
+
+func (s *Server) Append(r Runner) {
+	if s.Servers != nil {
```

Comment:
needed? wanted?

Reply:
Makes for a cleaner interface. I would leave it as is.